app/src/main/java/org/wikipedia/diff/ArticleEditDetailsFragment.kt

```diff
@@ -0,0 +1,419 @@
+                    sendThanks()
+                }
+                .setNegativeButton(R.string.thank_dialog_negative_button_text, null)
+                .create()
+        dialog.layoutInflater.inflate(R.layout.view_thank_dialog, parent)
+        dialog.setOnShowListener {
+            dialog.getButton(AlertDialog.BUTTON_NEGATIVE)
+                    .setTextColor(ResourceUtil.getThemedColor(requireContext(), R.attr.secondary_text_color))
+        }
+        dialog.show()
+    }
+
+    private fun sendThanks() {
+        ServiceFactory.get(WikiSite.forLanguageCode(languageCode)).csrfToken
+                .subscribeOn(Schedulers.io())
+                .flatMap {
+                    ServiceFactory.get(WikiSite.forLanguageCode(languageCode)).postThanksToRevision(revisionId, it.query()!!.csrfToken()!!)
+                }
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe({
+                    FeedbackUtil.showMessage(activity, getString(R.string.thank_success_message, username))
+                    setButtonTextAndIconColor(thankButton, ResourceUtil.getThemedColor(requireContext(),
+                            R.attr.material_theme_de_emphasised_color))
+                    thankButton.isClickable = false
+                })
+                { Consumer { t: Throwable? -> L.e(t) } }
+    }
+
+    private fun fetchDiffText() {
+        disposables.add(ServiceFactory.getCoreRest(WikiSite.forLanguageCode(languageCode)).getDiff(olderRevisionId, revisionId)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe({
+                    displayDiff(it.diffs)
+                }) { t: Throwable? -> L.e(t) })
+    }
+
+    private fun displayDiff(diffs: MutableList<DiffItem>) {
+        for (diff in diffs) {
+            val prefixLength = diffText.text.length
+            val foregroundAddedColor = ForegroundColorSpan(ResourceUtil.getThemedColor(requireContext(), R.attr.color_group_64))
+            val foregroundRemovedColor = ForegroundColorSpan(ResourceUtil.getThemedColor(requireContext(), R.attr.color_group_65))
+            val boldStyle = StyleSpan(Typeface.BOLD)
+
+            val spannableString = SpannableStringBuilder(diffText.text).append(if (diff.text.isNotEmpty()) diff.text else "\n")
+            when (diff.type) {
+                DIFF_TYPE_LINE_WITH_SAME_CONTENT -> {
+                }
+                DIFF_TYPE_LINE_ADDED -> {
+                    spannableString.setSpan(BackgroundColorSpan(ResourceUtil.getThemedColor(requireContext(),
+                            R.attr.edit_green_highlight)), prefixLength, prefixLength + diff.text.length, 0)
+                    spannableString.setSpan(boldStyle, prefixLength, prefixLength + diff.text.length, 0)
+                    spannableString.setSpan(foregroundAddedColor, prefixLength, prefixLength + diff.text.length, 0)
```

Comment:
Just curious that is it possible to make these actions into a method, which may be cleaner and easier to understand?